server/server.js

```diff
@@ -71,32 +74,101 @@ let videos = [
   },
 ];
 
+//GET "/"
+// app.get("/", (req, res) => {
+// Delete this line after you've confirmed your server is running
+//   res.send(videos);
+// });
+
 // GET "/"
-app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send(videos);
+app.get("/videos", async (req, res) => {
+  try {
+    const allVideos = await pool.query("SELECT * FROM videos");
+    res.json(allVideos.rows);
+  } catch (err) {
+    console.error(err.message);
+  }
 });
-app.post("/", (req, res) => {
-  let data = req.body;
-  res.status(200).send(...videos, JSON.stringify(data));
+
+// app.post("/videos", (req, res) => {
+//   let data = req.body;
+//   res.status(200).send(...videos, JSON.stringify(data));
+// });
+
+// POST a new Video
+
+app.post("/videos", async (req, res) => {
+  try {
+    const { title, url } = req.body;
+    if (!title || !url) {
+      return res
+        .status(400)
+        .json({ result: "failure", message: "Title and URL are required" });
+    }
+    const newVideo = await pool.query(
+      "INSERT INTO VIDEOS (title,url,rating) VALUES($1, $2, $3) RETURNING *",
+      [title, url, 0]
+    );
+    res.json(newVideo.rows[0]);
+  } catch (err) {
+    console.log(err.message);
```

Comment:
If you're handling the error, make sure to return a value to the user `res.status(x).json({data: null, error: err})` or something like that